internal/service/currency-service.go

```diff
@@ -9,49 +9,52 @@ import (
 	"net/url"
 	"time"
 
+	"genesis-currency-api/pkg/config"
+
 	"genesis-currency-api/pkg/dto"
 	"genesis-currency-api/pkg/errors"
 	"genesis-currency-api/pkg/util/date"
-	"github.com/spf13/viper"
 )
 
-type CurrencyService struct {
+type CurrencyServiceImpl struct {
```

Comment:
And after, if you name interface `CurrencyUpdater` you can name this `CurrencyService`. 
If you put interface in the package that uses values of the interface type, you will not have name collision.